pkg/apis/operator/v1alpha1/tektonpipeline_lifecycle.go

```diff
@@ -17,111 +17,155 @@ limitations under the License.
+func (tps *TektonPipelineStatus) MarkPostReconcilerComplete() {
+	pipelineCondSet.Manage(tps).MarkTrue(PostReconciler)
+}
+
+func (tps *TektonPipelineStatus) MarkNotReady(msg string) {
 	pipelineCondSet.Manage(tps).MarkFalse(
-		DeploymentsAvailable,
-		"NotReady",
-		"Waiting on deployments")
+		apis.ConditionReady,
+		"Error",
+		"Ready: %s", msg)
 }
 
-// MarkDependenciesInstalled marks the DependenciesInstalled status as true.
-func (tps *TektonPipelineStatus) MarkDependenciesInstalled() {
-	pipelineCondSet.Manage(tps).MarkTrue(DependenciesInstalled)
+func (tps *TektonPipelineStatus) MarkPreReconcilerFailed(msg string) {
+	tps.MarkNotReady("PreReconciliation failed")
+	pipelineCondSet.Manage(tps).MarkFalse(
+		PreReconciler,
+		"Error",
+		"PreReconciliation failed with message: %s", msg)
 }
 
-// MarkDependencyInstalling marks the DependenciesInstalled status as false with the
-// given message.
-func (tps *TektonPipelineStatus) MarkDependencyInstalling(msg string) {
+func (tps *TektonPipelineStatus) MarkInstallerSetNotAvailable(msg string) {
+	tps.MarkNotReady("TektonInstallerSet not ready")
 	pipelineCondSet.Manage(tps).MarkFalse(
-		DependenciesInstalled,
-		"Installing",
-		"Dependency installing: %s", msg)
+		InstallerSetAvailable,
+		"Error",
+		"Installer set not ready: %s", msg)
 }
 
-// MarkDependencyMissing marks the DependenciesInstalled status as false with the
-// given message.
-func (tps *TektonPipelineStatus) MarkDependencyMissing(msg string) {
+func (tps *TektonPipelineStatus) MarkInstallerSetNotReady(msg string) {
+	tps.MarkNotReady("TektonInstallerSet not ready")
+	pipelineCondSet.Manage(tps).MarkFalse(
+		InstallerSetReady,
+		"Error",
+		"Installer set not ready: %s", msg)
+}
+
+func (tps *TektonPipelineStatus) MarkPostReconcilerFailed(msg string) {
+	tps.MarkNotReady("PostReconciliation failed")
 	pipelineCondSet.Manage(tps).MarkFalse(
-		DependenciesInstalled,
+		PostReconciler,
 		"Error",
-		"Dependency missing: %s", msg)
+		"PostReconciliation failed with message: %s", msg)
+}
+
+// TODO: below methods are not required for TektonPipeline
+// but as extension implements TektonComponent we need to defined them
+// this will be removed
+
+func (tps *TektonPipelineStatus) GetTektonInstallerSet() string {
+	return tps.TektonInstallerSet
+}
+
+func (tps *TektonPipelineStatus) SetTektonInstallerSet(installerSet string) {
+	tps.TektonInstallerSet = installerSet
 }
 
-// GetVersion gets the currently installed version of the component.
 func (tps *TektonPipelineStatus) GetVersion() string {
 	return tps.Version
 }
 
-// SetVersion sets the currently installed version of the component.
 func (tps *TektonPipelineStatus) SetVersion(version string) {
 	tps.Version = version
 }
 
-// GetManifests gets the url links of the manifests.
-func (tps *TektonPipelineStatus) GetManifests() []string {
-	return tps.Manifests
+func (tps *TektonPipelineStatus) MarkInstallSucceeded() {
```

Comment:
this are not used now but we need to define as `extension` uses `TektonComponent` 
but will be removed as refactor other components refactor the TektonComponent